### ISSUE TITLE
can write but not read on local mapped drive

### ISSUE THREAD

**Issue by hspantera** [openzfsonwindows/openzfs#365]
Hope someone can help me. configured pool with assing letter on windows 11. can acess from the local pc to the drive letter and can read,write and make everything. but when i share the drive letter to can access from another pc on the same network i can acess the share see whats inside, write to but i cant read nothing froom there. i have some movies that i can see on my netwok share but when i try to open vlc says the file dont exist and cant reproduce. but i can write to that share. i give all permissionson windows pc. do i have to give any permissions on openzfs to acess? hope someone can help.

### System information

Type | Version/Name
 --- | ---
Distribution Name	|
Distribution Version	|
Kernel Version	|
Architecture	|
OpenZFS Version	|

**Comment by lundman** [comment]
OK, so everything works as expected on local machine.
Then you share over Samba
A network client accessing it, can read, but not write ?

**Comment by hspantera** [comment]
Yes . On local machine 100%. Local share can write but not read.. strange . I can see the files . For example I see all my Mkv movies but when I open gives me an error. Don’t play.

**Comment by hspantera** [comment]
Sorry. On network client that this happens. Can’t read but can write.

**Comment by lundman** [comment]
Yeah, I'll dig into it tomorrow.
If you are bored, run "FileSpy" as Administrator http://www.zezula.net/en/tools/filespy.html and change it to monitor your ZFS drive, then make the client write problem happen, could be it will give hints.

**Comment by hspantera** [comment]
No.. no problem in write. Only read.
![IMG_6747](https://github.com/openzfsonwindows/openzfs/assets/14846723/d017ecd5-4644-45fc-876f-2f5b6634952d)

this is my local zigs drive.

![IMG_6748](https://github.com/openzfsonwindows/openzfs/assets/14846723/59a3605b-8273-469b-87a5-71a469a446d5)


This is my network share on another windows 11 pc on same network

![IMG_6749](https://github.com/openzfsonwindows/openzfs/assets/14846723/eb1dace6-3452-4218-aefb-bbc9f5498f82)
![IMG_6750](https://github.com/openzfsonwindows/openzfs/assets/14846723/946ab1e3-b6c5-4464-88fa-8d94f6c5f393)

I can see all files and folders . Can write to share but can’t read. If I try to read from voc it show me this error.

![IMG_6751](https://github.com/openzfsonwindows/openzfs/assets/14846723/b398c2ad-78e2-4ece-8eef-2583986815f7)

very strange. Because I can access.. see what’s in there.. write to it. But can’t read files.

**Comment by lundman** [comment]
error when dragging to desktop or somewhere to copy?

**Comment by hspantera** [comment]
Yes.. won’t let me read too

**Comment by lundman** [comment]
ok thanks. I'll check what FileSpy says tomorrow

**Comment by hspantera** [comment]
[FileSpy info.zip](https://github.com/openzfsonwindows/openzfs/files/14731221/FileSpy.info.zip)

here is filespy when i try to acess video file on client share

**Comment by lundman** [comment]
Hmm interesting - it's certainly trying a few times, and something makes it unhappy.

**Comment by hspantera** [comment]
very strange. but it happens on both windows clients with windows 11 pro workstation installed. same error on both.

**Comment by lundman** [comment]
Refuse a lock, and
the open-by-id seems to fail. 
Could be either of those two, or of course, something completely different

**Comment by hspantera** [comment]
i have 2 windows clients on same network and they both do exactly the same thing. can see whats inside, cna write to the share but cant read or do nothing with existing files.

**Comment by hspantera** [comment]
Another strange thing that happens

![IMG_6752](https://github.com/openzfsonwindows/openzfs/assets/14846723/07b95bf7-5152-49c8-b779-0f56cc39a2c4)

open a folder 

![IMG_6753](https://github.com/openzfsonwindows/openzfs/assets/14846723/4c900c17-604c-485f-a114-b05ff8f84c61)

then select mkv file 

![IMG_6754](https://github.com/openzfsonwindows/openzfs/assets/14846723/84ff5534-0bd6-4a1c-8313-be176466110b)

give me an error on vlc can’t read

![IMG_6755](https://github.com/openzfsonwindows/openzfs/assets/14846723/0dbeef0d-dab0-4014-8f95-d9e8333175fa)

and if I open again same folder it shows empty. No mkv file. But I go to the original pc and open folder and it’s there. Only on client don’t show. It does this with all files I try to open on client share. Open file. Gives error. Open folder again and it’s empty. But the file still existed on original pc.

**Comment by hspantera** [comment]
But after a while I open the folder and it shows the file again. But if I try to play gives error and desapears again.

**Comment by hspantera** [comment]
![IMG_6757](https://github.com/openzfsonwindows/openzfs/assets/14846723/daf53177-98bd-40c6-a11b-c003d4c98783)

this is my zpool.

**Comment by lundman** [comment]
yes it is just that it uses something unexpected to verify the file is there, and ZFS replies incorrectly, so it removes it. Small files, like .nfo/.sfv also fails to read? In case it is a filesize thing

**Comment by hspantera** [comment]
It opens a small winrar file of file spy for example. But .srt file don’t show on client.
I copy 5 srt files to the original zpool folder and on client don’t show the files.

**Comment by hspantera** [comment]
.txt file don’t show too.

**Comment by lundman** [comment]
OK, not a filesize thing. I didn't it was, but worth ruling out.

**Comment by hspantera** [comment]
Hi. Have a question not related with this problem. I extended my raidz2 with one new hard drive. Has you can see in picture is resizing. It says it will take 4 days. My question is:
Can I export the pool and change all drives to new system and import the pool and continue to resizing from where it lefted when I export it? Thanks

![IMG_6765](https://github.com/openzfsonwindows/openzfs/assets/14846723/2924f6aa-2bbd-4f9a-8603-7b17ab774146)

**Comment by lundman** [comment]
ZFS it supposed to let you do that, and it will continue to resilver. Haven't tried it but the core is the same.

**Comment by hspantera** [comment]
Ok. I will do it.. after I import in new system I have to execute any command? Or it starts resizing automatically?

**Comment by lundman** [comment]
Should just start, afaik

**Comment by hspantera** [comment]
Yes.. worked perfectly.. started resizing automatically. Thanks. What about the bug of share? Did you see anything? Thanks.

**Comment by lundman** [comment]
OK, so smb works here, but one side is Win10 - so it could be that Win11 has changed something. But in the FileSpy it clearly tries FileFsObjectIdInformation multiple times, so I have implemented that - perhaps it will help.

Can you try:

OpenZFSOnWindows-debug-2.2.3rc2-2-geb5def3b01.exe

from releases.

**Comment by hspantera** [comment]
Installed and still has same problem..

**Comment by lundman** [comment]
Rats - can you do a new FileSpy - so we can see how it behaves now?

**Comment by hspantera** [comment]
[FileSpylog.zip](https://github.com/openzfsonwindows/openzfs/files/14741524/FileSpylog.zip)

here t is.

**Comment by lundman** [comment]
Thanks. Looks pretty good, does retry a bunch of times, which sort of suggests it's not happy about something, even if we now have implemented one of the calls.
IOCTL_MOUNTDEV_QUERY_DEVICE_NAME still fail, but I think it is supposed to.
I will have to set up a Win11 client as well, and give it a whirl myself.

**Comment by lundman** [comment]
If you are reeeaaallly bored, and I don't expect you to be, create a .vhdx file, format NTFS, copy a file to it - perhaps the same one as in filespy, and run FileSpy reading over SMB from that NTFS device. Can compare the two filespy logs and see if anything stands out.

but i get it if you dont have time, I'll get to it.

**Comment by lundman** [comment]
Hmm even with both VMs being Win11 I can not replicate the read problem. I get plenty of permission issues over SMB, like trying to create a new directory, but playing the files seem to work fine.

**Comment by hspantera** [comment]
Maybe it’s because it’s windows 11 pro workstation?

**Comment by hspantera** [comment]
All of my 3 systems are windows 11 pro workstation.

**Comment by lundman** [comment]
Hard to imagine why they would have a different SMB on it, but I suppose it is possible.

**Comment by sskras** [comment]
@hspantera: you probably know – there are two permission mechanisms that control shared folders:
[NTFS Permissions vs. Share Permissions – What’s the Difference?](https://www.tenfold-security.com/en/ntfs-permissions-and-share-permissions-whats-the-difference/#:~:text=NTFS%20permissions%20govern%20access%20to,is%20made%20via%20the%20network.)

What's both of your sets? Mine:

<img src="https://github.com/openzfsonwindows/openzfs/assets/7887758/ddf497c4-7a0a-4c41-afb4-89ed90d74f84" width="75%">

**Comment by hspantera** [comment]
I activated share via windows and get full permissions for all.. so it’s anything I have to do? I have a normal hard drive with this configuration and it works well.. only the pool share ( that I give full permitions via windows) has this problem.

**Comment by sskras** [comment]
To be double sure, can you please get the output of `powershell "Get-SmbShareAccess -Name SHARE-NAME"` for both of you shares?

**Comment by hspantera** [comment]
![IMG_6770](https://github.com/openzfsonwindows/openzfs/assets/14846723/3722c26c-5922-4ce3-bbb6-23fe8e5eb711)

it gives me this error.

**Comment by lundman** [comment]
oohh what does ntfs share say? can you filespy that too and I'll implement whatever is missing

**Comment by hspantera** [comment]
But I make a shed not by command line. I make it on windows normally.

**Comment by hspantera** [comment]
I’m running locally. So a shared normally as I shared normal folders on pc.

**Comment by hspantera** [comment]
![IMG_6771](https://github.com/openzfsonwindows/openzfs/assets/14846723/44a335d4-ac11-4d5a-adbe-6aae1890d18b)

my pool share is X:
How do I execute the command for knowing the permitions of NTFS and share permissions?

**Comment by lundman** [comment]
share one of your ntfs drive (or make a vhdx) then map it to drive letter

**Comment by hspantera** [comment]
![IMG_6772](https://github.com/openzfsonwindows/openzfs/assets/14846723/8a942f47-ba34-4011-9719-319f841a99d2)

my pool share permitimos NTFS.

**Comment by hspantera** [comment]
powershell "Get-SmbShareAccess -Name X”
On cmd don’t work.
![IMG_6773](https://github.com/openzfsonwindows/openzfs/assets/14846723/043caa82-07d3-450d-a49a-afb351eb6c7a)

I’m doing something wrong?

**Comment by hspantera** [comment]
Already dit it.

![IMG_6774](https://github.com/openzfsonwindows/openzfs/assets/14846723/ccc9b7b6-9d79-4eae-92dc-dcb57439bc76)

I have full permitions.

**Comment by sskras** [comment]
> Already dit it.
  [...]
> I have full permitions.

Full Access is permitted via Network only for the `Todos` account, and **no one else**.
Are you sure you are accessing this share from the remote machine (via SMB) under **the same username**?

PS. Also can you please return to the NTFS permissions command and expand it to that form:
```
Get-Acl D: | Format-List
```
The default output looks truncated in your photo. It should look like this:
```
PS C:\Users\saukrs> Get-Acl X: | Format-List


Path   : Microsoft.PowerShell.Core\FileSystem::D:\
Owner  : NT AUTHORITY\SYSTEM
Group  : NT AUTHORITY\SYSTEM
Access : NT AUTHORITY\Authenticated Users Allow  -536805376
         NT AUTHORITY\Authenticated Users Allow  Modify, Synchronize
         NT AUTHORITY\SYSTEM Allow  FullControl
         NT AUTHORITY\SYSTEM Allow  268435456
         BUILTIN\Administrators Allow  268435456
         BUILTIN\Administrators Allow  FullControl
         BUILTIN\Users Allow  ReadAndExecute, Synchronize
         BUILTIN\Users Allow  -1610612736
Audit  :
Sddl   : O:SYG:SYD:(A;OICIIO;SDGXGWGR;;;AU)(A;;0x1301bf;;;AU)(A;;FA;;;SY)(A;OICIIO;GA;;;SY)(A;OICIIO;GA;;;BA)(A;;FA;;;B
         A)(A;;0x1200a9;;;BU)(A;OICIIO;GXGR;;;BU)
```

**Comment by hspantera** [comment]
![IMG_6775](https://github.com/openzfsonwindows/openzfs/assets/14846723/ad045361-d6b3-4b4c-b0b8-af6e1c713044)

**Comment by hspantera** [comment]
I run command on a: that is a shared folder on windows and it’s all the same. But I notice the group it’s different. Is that normal? 

this is z: windows local hard drive share
![IMG_6776](https://github.com/openzfsonwindows/openzfs/assets/14846723/d361b3a2-c3a2-4a8d-9c3a-195b8fd36723)

this is x: zpool share

![IMG_6775](https://github.com/openzfsonwindows/openzfs/assets/14846723/10b786b3-d3b3-4f72-9a21-01c3d8698cce)

**Comment by sskras** [comment]
> I run command on z: that is a shared folder on windows and it’s all the same. But I notice the group it’s different. Is that normal?

No ideas, additional digging is needed. Eg: 
https://www.tenforums.com/network-sharing/113298-having-trouble-taking-ownership-usb-network-drive.html

Can we also get this output for comparison?
```
Get-Acl Z: | Format-List
```

**Comment by hspantera** [comment]
![IMG_6776](https://github.com/openzfsonwindows/openzfs/assets/14846723/4dcc46a3-6e01-4ee1-afe6-e0e515b299d1)

**Comment by hspantera** [comment]
![IMG_6777](https://github.com/openzfsonwindows/openzfs/assets/14846723/143d1f6d-482b-4639-9b52-314ae6137dff)

**Comment by hspantera** [comment]
No need for password. On both.

**Comment by sskras** [comment]
@lundman, please excuse me for intervening into your responses. It may be useful to repeat your last comments that I interrupted.

**Comment by lundman** [comment]
Quite interesting, since we have to convert Windows user SID to a Unix UID, and back, it is entirely possible that there could be issues there, do we return the same SID as was stored? Who knows, are there tools to test that?

**Comment by lundman** [comment]
OK so it turns out that `FILE_OPEN_BY_FILE_ID` can be called with a 16 byte `ObjectID` (Instead of the expected 8 byte `FileID`).

I have cut OpenZFSOnWindows-debug-2.2.3rc2-3-g0e4c16e0a6.exe with a fix for that.

**Comment by hspantera** [comment]
same problem with new version OpenZFSOnWindows-debug-2.2.3rc2-3-g0e4c16e0a6.exe

heres the log

[FileSpy log.zip](https://github.com/openzfsonwindows/openzfs/files/14785686/FileSpy.log.zip)

**Comment by lundman** [comment]
Ah shame, not a single failure in there now - so its a bit more obscure.

**Comment by lundman** [comment]
Yeha, its trying to read a locale file to print the output in whatever language you have set, but there is no locale for that one. So not a problem itself, possibly means it has the error by now and wants to know how to print it in locale language.

**Comment by hspantera** [comment]
new log file
[FileSpy log.zip](https://github.com/openzfsonwindows/openzfs/files/14785973/FileSpy.log.zip)

**Comment by lundman** [comment]
`IRP_MJ_FILE_SYSTEM_CONTROL/IRP_MN_KERNEL_CALL	00000054`
I'll check what that is, but its only once

**Comment by lundman** [comment]
FILE_DEVICE_POINT_OF_SERVICE

**Comment by hspantera** [comment]
Any news? Thanks.

**Comment by hspantera** [comment]
installed new version and same error. very strange. dont understand. everything works fine. super faste reads and writes. only this share problem.

**Comment by hspantera** [comment]
one question. to install new version i can download the executable file and run it right? dont need to delete the previous version?

**Comment by sskras** [comment]
> installed new version and same error. very strange. dont understand. everything works fine. super faste reads and writes. only this share problem.

I wonder if this is due to some change introduced by Windows 11. I'd suggest installing a different version (maybe in a VM) and try the driver here. Eg:

- [Windows 10 22H2](https://archive.org/details/en-us_windows_10_consumer_editions_version_22h2_updated_feb_2023_x64_dvd_c29e4bb3) (includes Windows 10 Pro for Workstations)
  (the checksum can be found [on Reddit](https://www.reddit.com/r/MSCDB/comments/119870l/windows_iso_updated_february_2023/))

... or maybe some earlier and simpler builds:

- [Windows 10 2004 English 64-bit [19041.264]](https://archive.org/details/Win10_2004_English_x64)
- [Windows 10 1703 English 64-bit [15063.0]](https://archive.org/details/Win10_1703_English_x64)
- [Windows 10 1909 English 64-bit [18363.418]](https://archive.org/details/Win10_1909_English_x64)
- [Windows 10 1507](https://archive.org/details/win-10-1507-english-x-64), the first version of Windows 10 released ever.

**Comment by hspantera** [comment]
Must be windows 11 workstation. From my iMac in same network can read and write normally..

**Comment by sskras** [comment]
macOS SMB client implementation and the configuration is too different to be compared.

What number does the `ver` or `winver` commands say on your client w11 machines?

**Comment by hspantera** [comment]
![IMG_6912](https://github.com/openzfsonwindows/openzfs/assets/14846723/046de415-bacd-488e-9011-481bb7d8b9fc)

it’s windows 11 workstation

**Comment by sskras** [comment]
@hspantera, please downscale your photos to save cloud disk space and everyone's traffic in future:)

> it’s windows 11 workstation

It seems I misunderstood a detail about your setup. So **Windows 11 Pro for Workstations** runs **both** on the server machine and on two client machines, right?

**Comment by hspantera** [comment]
Yes.. all my 3 pcs have windows 11 pro for workstation. Sorry for the fotos. I upload from my iPhone.

**Comment by sskras** [comment]
So your w11 build number is [23H2 [22631.3447]](https://learn.microsoft.com/en-us/windows/release-health/windows11-release-information#servicing-channels) released two days ago.

If you hesitate to try mounting that share using w10, maybe you should try lower versions of w11 instead?

- [Windows 11 21H2 English 64-bit [22000.194]](https://archive.org/details/Win11_English_x64)
- [Windows 11 21H2 English 64-bit [22000.318]](https://archive.org/details/Win11_English_x64v1)
- [Windows 11 22H2 English 64-bit Build-22622](https://archive.org/details/windows11_20220930)

**Comment by lundman** [comment]
I will grab a 11proWS VM and try it here - be useful to know if only one side (server, or client) needs to be 11proWS tho.

**Comment by lundman** [comment]
So then https://developer.microsoft.com/en-us/windows/downloads/virtual-machines/ should also have the issue? 
2403, Enterprise.

**Comment by lundman** [comment]
Hmm my dev target is already Win 11 Pro N - 23H2 [22635.2921]
So maybe that isn't it. I run ZFS on it, and smb to local-machine-ip so both smb client and server are on same machine - hoping smb isn't smart enough to do a short cut around it.

**Comment by lundman** [comment]
OK, connected to ZFS and mapped to Z: over loopback - checking the smb data

```
# Check if SMB1 is enabled (powershell, administrator)

PS C:\WINDOWS\system32> Get-WindowsOptionalFeature -Online -FeatureName SMB1Protocol


FeatureName      : SMB1Protocol
DisplayName      : SMB 1.0/CIFS File Sharing Support
Description      : Support for the SMB 1.0/CIFS file sharing protocol, and the Computer Browser protocol.
RestartRequired  : Possible
State            : Enabled
CustomProperties :
                   ServerComponent\Description : Support for the SMB 1.0/CIFS file sharing protocol, and the Computer
                   Browser protocol.
                   ServerComponent\DisplayName : SMB 1.0/CIFS File Sharing Support
                   ServerComponent\Id : 487
                   ServerComponent\Type : Feature
                   ServerComponent\UniqueName : FS-SMB1
                   ServerComponent\Deploys\Update\Name : SMB1Protocol



# Check if SMB2 is enabled

PS C:\WINDOWS\system32> Get-SmbServerConfiguration | Select EnableSMB2Protocol

EnableSMB2Protocol
------------------
              True


# Check what version the current connections are using

PS C:\WINDOWS\system32> Get-SmbConnection

ServerName ShareName UserName               Credential             Dialect NumOpens
---------- --------- --------               ----------             ------- --------
win11dev   IPC$      Win11Dev\WDKRemoteUser Win11Dev\WDKRemoteUser 3.1.1   0
win11dev   tools     Win11Dev\WDKRemoteUser Win11Dev\WDKRemoteUser 3.1.1   2

# More details

PS C:\WINDOWS\system32> Get-SmbConnection | Select-Object -Property *


SmbInstance           : Default
ContinuouslyAvailable : False
Credential            : Win11Dev\WDKRemoteUser
Dialect               : 3.1.1
Encrypted             : False
NumOpens              : 2
Redirected            : False
ServerName            : win11dev
ShareName             : tools
Signed                : False
UserName              : Win11Dev\WDKRemoteUser
PSComputerName        :
CimClass              : ROOT/Microsoft/Windows/SMB:MSFT_SmbConnection
CimInstanceProperties : {ContinuouslyAvailable, Credential, Dialect, Encrypted...}
CimSystemProperties   : Microsoft.Management.Infrastructure.CimSystemProperties
```

Looks like I am using SMB3

**Comment by andrewc12** [comment]
@andrewc12 reminder to create test for whatever this is.
Non working version is
https://github.com/openzfsonwindows/openzfs/tree/zfswin-2.2.3rc3

**Comment by hspantera** [comment]
Did it work?

**Comment by lundman** [comment]
My smb access is fine, no issues. Perhaps the broken one is using SMB1?

**Comment by hspantera** [comment]
How can I activate smb3? It’s possible on windows 11 pro workstation?

**Comment by lundman** [comment]
Hoping you would run those commands so we can see what version you are using

**Comment by hspantera** [comment]
What commands I have to do? Can you help me?

**Comment by lundman** [comment]
At the very least, run PowerShell as Administrator and run ` Get-SmbConnection | Select-Object -Property *`

**Comment by hspantera** [comment]
![IMG_6920](https://github.com/openzfsonwindows/openzfs/assets/14846723/79188501-e6c9-4905-abcd-1914dcdc153e)

**Comment by hspantera** [comment]
I think it’s 3.1.1

**Comment by hspantera** [comment]
I make all tests you said in the smb1/smb2/smb3 and smb1 was disable.. should I enable it? Smb2 and 3 where enable.

**Comment by lundman** [comment]
OK so that's not it then. Hmmm, what else could be different between us. You are on a different locale, but I assume you still have problems even if all names are straight ASCII ?

**Comment by hspantera** [comment]
It’s very strange.. it should work. But always same error trying to opening video files.. or some file.. it says “can’t open location file” but I can have access to files.. very strange..

**Comment by hspantera** [comment]
More strange is that I open shared zigs folder on other windows 11 pro workstation and I can copy from the shared folder to the local pc that I’m accessing.. just don’t read . But I can copy from shares folder???  How strange is that???

**Comment by hspantera** [comment]
Yes

**Comment by lundman** [comment]
oh, you said you can't read, so I've been testing with dragging files from smb to desktop. That reads the files. I can try playing with VLC as well then, make sure they start.

**Comment by hspantera** [comment]
I can drag and drop too.. but opening with vlc gives error.. or any player ..

**Comment by hspantera** [comment]
But another problem.. I have a folder with 10 mkt files.. I drag and drop one.. everything works fine.. but if I go to the other mkv file in that folder gives me an error of “the item it’s not located in this folder anymore” and I can’t xopy any more files from that folder. But if I go to another folder it lets me always copy one file and if I try the second one it always give me that error. Do this in every folders..

**Comment by hspantera** [comment]
Hello.. still no fix for mapped openzfs file not reading files?

**Comment by lundman** [comment]
I have not yet found a way to replicate this locally - it just works (unless I misunderstand what the problem is). Trying to think of other ways this can be added. Do you have space for VMs?

**Comment by hspantera** [comment]
Did you test on windows 11 pro workstation? Or just windows 11 pro?

**Comment by lundman** [comment]
My VM is https://developer.microsoft.com/en-us/windows/downloads/virtual-machines/

Point me to a version that you know fails?

**Comment by hspantera** [comment]
This is my windows https://www.microsoft.com/en-us/windows/business/windows-11-pro-workstations

**Comment by sskras** [comment]
It's not a VM neither an `.iso` file. The nearest thing I found was the `.iso`:
https://www.minitool.com/news/download-windows-11-pro-for-workstations.html#download-windows-11-pro-for-workstations-42952

![](https://images.minitool.com/minitool.com/images/uploads/2024/03/download-windows-11-pro-for-workstations-1.png)

**Comment by hspantera** [comment]
Im not using vm. Using the version installed on pc. Windows 11 pro workstation.

**Comment by sskras** [comment]
@hspantera commented [35 minutes ago](https://github.com/openzfsonwindows/openzfs/issues/365#issuecomment-2114328584):

> Maybe that’s the problem. You can’t replicate windows 11 pro workstation on VM.

Have you tried to replicate that in a VM, and the issue disappeared? That would help testing.

**Comment by hspantera** [comment]
Hi.. I have a problem.. I have this pool
![IMG_7174](https://github.com/openzfsonwindows/openzfs/assets/14846723/4cc9348e-bf05-4a0c-9349-a53225351a83)
And it’s not phisycal drive 1 in it. 
I want to make another zpool but it say that physical disk1 is already a pool enter. But it isn’t. I only have one pool and physical disk 1 it’s not in that pool. How I remove physical disc1 from pool member? Because it don’t belong to any pool.
![IMG_7175](https://github.com/openzfsonwindows/openzfs/assets/14846723/5c3e46eb-4dd6-402b-9a98-62ac29b0b279)
![IMG_7176](https://github.com/openzfsonwindows/openzfs/assets/14846723/cad6a671-ee78-488a-9224-64cee8536c1d)
Can you help me? Because when I try to make a new pool it doesn’t show option to select physical disc 1. And physical disc one don’t belong to the already existing pool. So physical disc one said it’s a pool member but it’s not from any pool. Can I remove it? So I can add it to new pool?

**Comment by sskras** [comment]
@hspantera commented [4 hours ago](https://github.com/openzfsonwindows/openzfs/issues/365#issuecomment-2118609113):

> Hi.. I have a problem.. I have this pool 

  [...]

> Can you help me?

IMO this is a separate issue.

@hspantera, I suggest to copy source of your message (by clicking Edit and then selecting all the editable text) and creating a new topic in the [Discussions](https://github.com/openzfsonwindows/openzfs/discussions) tab using the same text.

This issue is about file I/O failures on mapped drive.

**Comment by lundman** [comment]
Perhaps zpool labelclear on physicaldrive1 is needed? Or create it without the gui.

**Comment by hspantera** [comment]
Sorry. But I’m kind of new here and it was just a question. Solved the problem. Didn’t started my pool and all drives where free. So I don’t use the ones that already where assigned to my already created pool and it worked. And I have one more question. You are being so helpful. I’m trying to update napp-it but on newer version I get this error. Can’t do anything. Always give me this error everytime . Think you can help me? 
![IMG_7182](https://github.com/openzfsonwindows/openzfs/assets/14846723/4e21732d-80ea-406c-a15c-56123a809c23)
 It show me all options but always that error. And I read somewhere that you where realeasing a web gui . It’s already out?

**Comment by lundman** [comment]
For nappit we might need to invite napp-it's @gea but it is concerning the reddit user is deleted, and the windows.pdf is now 404. Wonder what happened there.

**Comment by hspantera** [comment]
But old version works ok. Don’t exist other web gui based?

**Comment by hspantera** [comment]
Sorry to bother again. To update to the rc4 version I have to rename this file to what?

![IMG_7185](https://github.com/openzfsonwindows/openzfs/assets/14846723/53be506e-3014-4f71-a532-197f336accb6)

rename …\openzfs.sys to for example …\teste.sys?

**Comment by hspantera** [comment]
Can’t just uninstall rc3 and install rc4?

**Comment by hspantera** [comment]
Hi. One question. I have a folder on my zás pool that I click and it won’t open. And if I try to delete it won’t delete. Any chance of tell me what I can do? It refuses to open and to delete. Only that folder.. I have 3000 folders. Only that one don’t open. If it’s error of the pool what I can do to solve this? Thanks.

**Comment by lundman** [comment]
Renaming? Best would be to look in cbuf after it failed, but that is difficult if you aren't setup to develop.

**Comment by hspantera** [comment]
It lets me rename the file. But still don’t let me delete or open folder. Is my pool in danger?

**Comment by lundman** [comment]
I wouldn't have thought so, I thought maybe it had a name it couldn't translate. I assume you have rebooted to see if that helps.

**Comment by hspantera** [comment]
Already rebooted. It’s the same. There is some option to force delete?

**Comment by hspantera** [comment]
I think I’m going to have to find another solution for my media. Using openings on windows since the release of stable version and here is my thoughts. Pros- very fast writes. Very fast reads. Cons- with the time it’s going to corrupt a file here. Another there Until the pool it’s compromised. Still very buggy on the options like shares and how it delete files. It’s a great filesystem but for windows it’s yet very unreliable. I think it have potential to get the best filesystem in the future but for now it’s very dangerous having a 70terabite storage on openzfs. In any moment it will corrupt on file or a folder and you cont do nothing about that other than destroy pool and create a new one. Other issue for normal users like me it’s the commands to execute openzfs. Napp-it simplifies a little bit but still not a filesystem that it’s straight forward. Thanks for all support and I will looking for new releases and hope one day it becomes native like NTFS.
